lib/beacon_web/components/layouts.ex

```diff
@@ -26,6 +26,10 @@ defmodule BeaconWeb.Layouts do
   #{File.read!(phoenix_live_view_path) |> String.replace("//# sourceMappingURL=", "// ")}
   #{File.read!(beacon_js_path)}
   """
+  def render("app.js", %{conn: %{private: %{beacon: %{live_socket_path: live_socket_path}}}}) do
+    String.replace(@app_js, "/live", live_socket_path)
+  end
```

Comment:
We can make it simpler by just returning the private value and injecting it into the template :)


```suggestion
  def live_socket_path(conn) do
    conn.private.beacon.live_socket_path
  end
```

And then add the following at https://github.com/BeaconCMS/beacon/blob/ec5e80c58891c62156298419f6b4b32ed3320eac/lib/beacon_web/components/layouts/runtime.html.heex#L2

```heex
<html lang="en" phx-socket={live_socket_path(@conn)}>
```

That works because:

1 - The value injected into `conn.private.beacon.live_socket_path` has always a default value or something custom defined by user, see https://github.com/BeaconCMS/beacon/blob/ec5e80c58891c62156298419f6b4b32ed3320eac/lib/beacon/router.ex#L61

2 - The LiveView JS client already expectes that html attr to find where to connect to, see https://github.com/BeaconCMS/beacon/blob/ec5e80c58891c62156298419f6b4b32ed3320eac/priv/assets/js/app.js#L4

Reply:
@leandrocp excellent, I've commited the recommended changes.

Thanks!